### PR TITLE
[CLI] Consolidate exception handlers in CommandLineInterface #11732

### PR DESCRIPTION
**Pull request by s0sharma**
Removed all the generic handlers for things like "...", "std::exception", "util::Exception" or "InternalCompilerError" inside various functions of CommandLineInterface.cpp file and added The top-level handler inside main()

### CODE REVIEW

File: solc/main.cpp [hunk 1]
```diff
@@ -53,24 +53,34 @@ static void setDefaultOrCLocale()
...
+	{
+		setDefaultOrCLocale();
+		solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
+		if (!cli.parseArguments(argc, argv) || !cli.readInputFiles() || !cli.processInput() || !cli.actOnInput())
+			return 1;
+
+		return 0;
+	}
+	catch (boost::exception const& _exception || Exception const& _exception || InternalCompilerError const& _exception
+		|| smtutil::SMTLogicError const& _exception)
```
Comment (hrkrshnn) [inline solc/main.cpp]: Would this line compile?

File: test/solc/main.cpp [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (hrkrshnn) [inline test/solc/main.cpp]: The CI fails because this wasn't added for compilation. Can you remove this both locally and from the PR?

**Comment by hrkrshnn** [comment]
- The extra file `solc/1` should be removed.
- The tests `test/scripts/fixtures/smt_contract_with_mixed_newlines.sol`, `test/scripts/fixtures/smt_contract_with_crlf_newlines.sol` and `test/libsolidity/syntaxTests/license/license_crlf_endings.sol` did not effectively change. Please remove it from the PR.

File: solc/main.cpp (hunk 1 above)
Comment (s0sharma) [reply solc/main.cpp]: Let me change that.

File: solc/main.cpp [hunk 3]
```diff
@@ -53,24 +52,33 @@ static void setDefaultOrCLocale()
...
+	try
+	{
+		setDefaultOrCLocale();
+		solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
+		if (!cli.parseArguments(argc, argv) || !cli.readInputFiles() || !cli.processInput() || !cli.actOnInput())
+			return 1;
+
+		return 0;
+	}
+	catch (boost::exception const& _exception || Exception const& _exception || InternalCompilerError const& _exception || smtutil::SMTLogicError const& _exception)
```
Comment (hrkrshnn) [inline solc/main.cpp]: I meant that this `||` for exceptions shouldn't compile. You would probably have to write different catch blocks.

**Comment by s0sharma** [comment]
Created new pr [here](https://github.com/ethereum/solidity/pull/11762/files). Please check there

**Pull request closed without merging**